servicemesh-install/topics/tutorial-threescale-adapter.adoc

```diff
@@ -0,0 +1,147 @@
+metadata:
+  name: threescale
+  namespace: istio-system
+spec:
+  match: destination.labels["service-mesh.3scale.net"] == "true"
+  actions:
+  - handler: threescale.handler.istio-system
+    instances:
+    - threescale-authorization
+----
+
+=== Routing service traffic through the adapter
+In order to drive traffic for your service through the adapter and be managed by 3scale, we need to match the rule `destination.labels["service-mesh.3scale.net"] == "true"` we previously created in the configuration, in the kind: rule resource.
+
+To do so, we need to add a label to the PodTemplateSpec on the Deployment of the target workload. For example, if we have a `productpage` service, whose Pod is managed by the `productpage-v1` deployment, by adding the above label under `spec.template.labels` in `productpage-v1`, we can have the adapter authorize requests to this service.
+
+
+= Configure the integration settings in 3scale
+[NOTE]
+====
+For 3scale SaaS customers, {ProductName} is enabled as part of the Early Access program.
+====
+
+=== Enabling service mesh for on-premises customers
+. Execute the following commands to enable service mesh in 3scale.
++
+```
+$ oc project <3scale-project>
+$ oc edit configmap system
+```
++
+. Edit the `rolling_updates.yml` file to add the correct value for `service_mesh_integration`.
++
+.rolling_updates.yml
+[source,yaml]
+----
+  production:
+    old_charts: false
+    new_provider_documentation: false
+    proxy_pro: false
+    instant_bill_plan_change: false
+    service_permissions: true
+    async_apicast_deploy: false
+    duplicate_application_id: true
+    duplicate_user_key: true
+    plan_changes_wizard: false
+    require_cc_on_signup: false
+    apicast_per_service: true
+    new_notification_system: true
+    cms_api: false
+    apicast_v2: true
+    forum: false
+    published_service_plan_signup: true
+    apicast_oidc: true
+    policies: true
+    proxy_private_base_path: true
+    service_mesh_integration: true
+----
++
+. Run the following command:
++
+```
+restart system-app, system-sidekiq pods
+```
+
+=== Integration settings
+. Navigate to *[your_API_name] > Integration > Configuration*.
+. A the top of the *Integration* page click on *edit integration settings* in the top right corner.
+. Under the *Service Mesh* heading, select the *Istio* option.
+. Scroll to the bottom of the page and click *Update Service*.
+
+=== Caching behavior
+Responses from 3scale System API's will be cached by default within the adapter. Entries will be purged from the cache when they become older than the `THREESCALE_CACHE_TTL_SECONDS` value. Also by default, automatic refreshing of cached entries will be attempted seconds before they expire, based on the `THREESCALE_CACHE_REFRESH_SECONDS` value. Automatic refreshing can be disabled by setting this value higher than the `THREESCALE_CACHE_TTL_SECONDS` value.
+
+Caching can be disabled entirely by setting `THREESCALE_CACHE_ENTRIES_MAX` to a non-positive value.
+
```

Comment:
It would be better if the above referenced the custom resource elements rather than environment variables which are not directly exposed to the users.  Anything beginning with THREESCALE is suspect.

Reply:
I've updated all of them to reflect the custom resource elements.